include/clang/Basic/MacroBuilder.h

```diff
@@ -27,18 +27,18 @@ class MacroBuilder {
   MacroBuilder(raw_ostream &Output) : Out(Output) {}
 
   /// Append a \#define line for macro of the form "\#define Name Value\n".
-  void defineMacro(const Twine &Name, const Twine &Value = "1") {
+  virtual void defineMacro(const Twine &Name, const Twine &Value = "1") {
     Out << "#define " << Name << ' ' << Value << '\n';
   }
 
   /// Append a \#undef line for Name.  Name should be of the form XXX
   /// and we emit "\#undef XXX".
-  void undefineMacro(const Twine &Name) {
+  virtual void undefineMacro(const Twine &Name) {
```

Comment:
Since you don't need to use `undefineMacro()` and `append()` in `FlangMacroBuilder` you probably don't need to make them virtual and provide a stub for them later.